src/Raven.Server/Documents/Handlers/IndexHandler.cs

```diff
@@ -600,14 +601,20 @@ public Task GetErrors()
         [RavenAction("/databases/*/indexes/terms", "GET", AuthorizationStatus.ValidUser)]
         public Task Terms()
         {
-            var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
             var field = GetQueryStringValueAndAssertIfSingleAndNotEmpty("field");
-            var fromValue = GetStringQueryString("fromValue", required: false);
 
             using (var token = CreateTimeLimitedOperationToken())
             using (Database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
             using (context.OpenReadTransaction())
             {
+                var name = GetStringQueryString("name", false) ?? GetIndexNameFromCollectionAndField(field, context);
+                if (name == null)
```

Comment:
This is a breaking change. We were throwing in that case.

Two exceptions here:
1. The one from GetQueryStringValueAndAssertIfSingleAndNotEmpty
2. When 'name' is specified (or 'collection') and we do not have an index then we are throwing IndexDoesNotExistException

Reply:
Check my other comment for point 1. and I'm good with point 2.